docs/recipes/add_dataset.md

```diff
@@ -0,0 +1,112 @@
+# Recipe: Adding a New Pretraining Dataset to Marin
+
+## Overview
+This recipe provides a structured guide for humans and AI agents to add new pretraining datasets to Marin. It emphasizes a step-by-step approach to ensure compatibility with Marin's data pipelines, such as tokenization and training workflows. Starting with schema inspection helps identify splits, subsets, and text fields, making the process more efficient and "roomba-friendly" (i.e., amenable to automation and junior contributors). These instructions also facilitate human onboarding by enforcing clear, reproducible practices.
+
+> **Reference:** Derived from community discussions on agent-friendly workflows, including @dlwh's checklist for dataset addition.
+
+## Prerequisites
+- Install the [Hugging Face `datasets` library](https://huggingface.co/docs/datasets/) (e.g., `pip install datasets`).
+- For YAML output, install `pyyaml` (e.g., `pip install pyyaml`).
+- Ensure access to the dataset: Hugging Face Hub ID, local path, or other supported formats.
+- Marin installed in editable mode (`pip install -e .`) for CLI access, or run the script standalone.
```

Comment:
can we simplify this to `uv run marin/tools/get_hf_dataset_schema.py` or something

Reply:
we use pyproject.toml which is more modern. `uv run` will automatically install dependencies etc.